### PR TITLE
DTSPB-3680 No Future Date for CW Caveat Application

### PR DESCRIPTION
**Pull request by MarcusHMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPB-3680

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/caseCreator/java/uk/gov/hmcts/probate/casecreator/util/TestCaseCreator.java [hunk 1]
```diff
@@ -179,6 +179,7 @@ private String generateClientToken() throws Exception {
     }
 
     private String generateClientCode() throws Exception {
+        log.info("generateClientCode idamUsername : {} --{}", idamUsername,idamPassword);
```
Comment (FeliTam) [inline src/caseCreator/java/uk/gov/hmcts/probate/casecreator/util/TestCaseCreator.java]: not sure what these changes about, are they for local testing purpose?

File: src/test/java/uk/gov/hmcts/probate/validator/CaveatDodValidationRuleTest.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    void shouldThrowDodExceptionRaiseCaveatValidate() {
+        CaveatDetails caveatDetails =
+                new CaveatDetails(caveatDataWithDodException, LAST_MODIFIED, CASE_ID);
+        Assertions.assertThatThrownBy(() -> {
+            caveatDodValidationRule.validate(caveatDetails);
+        })
+            .isInstanceOf(BusinessValidationException.class)
+            .hasMessage("Date of death cannot be in the future");
+    }
+
```
Comment (FeliTam) [inline src/test/java/uk/gov/hmcts/probate/validator/CaveatDodValidationRuleTest.java:80]: Can you also add test case for caveat on the same day as the DOD?

File: src/caseCreator/java/uk/gov/hmcts/probate/casecreator/util/TestCaseCreator.java (hunk 1 above)
Comment (MarcusHMCTS) [reply src/caseCreator/java/uk/gov/hmcts/probate/casecreator/util/TestCaseCreator.java]: done, removed

File: src/test/java/uk/gov/hmcts/probate/validator/CaveatDodValidationRuleTest.java (hunk 2 above)
Comment (MarcusHMCTS) [reply src/test/java/uk/gov/hmcts/probate/validator/CaveatDodValidationRuleTest.java:80]: added

**Review by IswaryaPepakayala — APPROVED** [review]

File: src/caseCreator/java/uk/gov/hmcts/probate/casecreator/util/TestCaseCreator.java [hunk 3]
```diff
@@ -71,10 +71,10 @@ public class TestCaseCreator {
     private RelaxedServiceAuthTokenGenerator relaxedServiceAuthTokenGenerator;
     private ObjectMapper objectMapper = new ObjectMapper().registerModule(new JavaTimeModule());
 
-    {
+    /*{
         System.setProperty("socksProxyHost", "localhost");
         System.setProperty("socksProxyPort", "9090");
-    }
+    }*/
```
Comment (DelythJustice) [inline src/caseCreator/java/uk/gov/hmcts/probate/casecreator/util/TestCaseCreator.java]: Should this be removed?

**Review by DelythJustice — APPROVED** [review]

File: src/caseCreator/java/uk/gov/hmcts/probate/casecreator/util/TestCaseCreator.java (hunk 3 above)
Comment (MarcusHMCTS) [reply src/caseCreator/java/uk/gov/hmcts/probate/casecreator/util/TestCaseCreator.java]: yes, removed

**Pull request merged**